### PR TITLE
Improve Code Quality through assertions

### PR DESCRIPTION
**Pull request by ZhangAnli**
Increased the number of assertions in view command and find command - and related files - to improve coding quality.

### CODE REVIEW

File: src/main/java/dog/pawbook/logic/parser/FindCommandParser.java [hunk 1]
```diff
@@ -27,6 +28,8 @@ public FindCommand parse(String args) throws ParseException {
         }
         String[] nameKeywords = trimmedArgs.split("\\s+");
         List<String> keywords = Arrays.asList(nameKeywords);
+
+        assert (!keywords.isEmpty());
```
Comment (kouyk) [inline src/main/java/dog/pawbook/logic/parser/FindCommandParser.java:32]: Makes more sense check this within `NameComtainsKeywordsPredicate`

File: src/main/java/dog/pawbook/logic/parser/ParserUtil.java [hunk 2]
```diff
@@ -190,7 +191,7 @@ public static int parseId(String id) throws ParseException {
         try {
             return Integer.parseInt(trimmedOwnerId);
         } catch (NumberFormatException e) {
-            throw new ParseException("ID must be a positive integer!");
+            throw new ParseException(Messages.MESSAGE_INVALID_ENTITY_ID);
```
Comment (kouyk) [inline src/main/java/dog/pawbook/logic/parser/ParserUtil.java:194]: just statically import, looks neater

File: src/main/java/dog/pawbook/logic/parser/ViewCommandParser.java [hunk 3]
```diff
@@ -30,6 +32,10 @@ public ViewCommand parse(String args) throws ParseException {
         } catch (NumberFormatException e) {
             throw new ParseException(MESSAGE_INVALID_ENTITY_ID);
         }
```
Comment (kouyk) [inline src/main/java/dog/pawbook/logic/parser/ViewCommandParser.java:34]: Violates DRY principle, can just use `ParserUtil.parseId` here

File: src/main/java/dog/pawbook/model/managedentity/IdMatchPredicate.java [hunk 4]
```diff
@@ -18,18 +19,21 @@
...
     public IdMatchPredicate(Collection<Integer> idList) {
+        assert (!idList.isEmpty());
         this.relatedIds.addAll(idList);
     }
 
     /**
      * Construct a predicate for a single ID.
      */
     public IdMatchPredicate(Integer id) {
+        assert(!id.equals(null));
```
Comment (kouyk) [inline src/main/java/dog/pawbook/model/managedentity/IdMatchPredicate.java:30]: `requireNonNull`

File: src/main/java/dog/pawbook/model/managedentity/IdMatchPredicate.java [hunk 5]
```diff
@@ -18,18 +19,21 @@
...
         this.relatedIds.add(id);
     }
 
     @Override
     public boolean test(Pair<Integer, Entity> idEntityPair) {
+        assert(!idEntityPair.equals(null));
```
Comment (kouyk) [inline src/main/java/dog/pawbook/model/managedentity/IdMatchPredicate.java:36]: `requireNonNull`

File: src/main/java/dog/pawbook/logic/parser/FindCommandParser.java (hunk 1 above)
Comment (ZhangAnli) [reply src/main/java/dog/pawbook/logic/parser/FindCommandParser.java:32]: Edited.

File: src/main/java/dog/pawbook/logic/parser/ParserUtil.java (hunk 2 above)
Comment (ZhangAnli) [reply src/main/java/dog/pawbook/logic/parser/ParserUtil.java:194]: Edited.

File: src/main/java/dog/pawbook/model/managedentity/IdMatchPredicate.java (hunk 4 above)
Comment (ZhangAnli) [reply src/main/java/dog/pawbook/model/managedentity/IdMatchPredicate.java:30]: Edited.

File: src/main/java/dog/pawbook/model/managedentity/IdMatchPredicate.java (hunk 5 above)
Comment (ZhangAnli) [reply src/main/java/dog/pawbook/model/managedentity/IdMatchPredicate.java:36]: Edited.

File: src/main/java/dog/pawbook/logic/parser/ViewCommandParser.java (hunk 3 above)
Comment (ZhangAnli) [reply src/main/java/dog/pawbook/logic/parser/ViewCommandParser.java:34]: Edited.

**Review by kouyk — APPROVED** [review]

**Pull request merged**